pkg/plugin/integration_test.go

```diff
@@ -0,0 +1,318 @@
+	})
+	cli.PrependReactor("get", "nodes", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		get := action.(k8stesting.GetAction)
+		node := &corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      get.GetName(),
+				Namespace: get.GetNamespace(),
+			},
+			Status: corev1.NodeStatus{
+				Conditions: []corev1.NodeCondition{
+					{
+						Type:   corev1.NodeReady,
+						Status: corev1.ConditionTrue,
+					},
+				},
+			},
+		}
+		return true, node, nil
+	})
+
+	cli.AddProxyReactor("services", func(action k8stesting.Action) (handled bool, ret restclient.ResponseWrapper, err error) {
+		return true, newFakeResponseWrapper(nil), nil
+	})
+
+	priv := securityapi.SecurityContextConstraints{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "securitycontextconstraints",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "privileged",
+		},
+		AllowPrivilegedContainer: true,
+		Users:                    []string{"system:admin"},
+	}
+	seccli := fakesec.NewSimpleClientset()
+	seccli.PrependReactor("get", "*", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		get := action.(k8stesting.GetAction)
+		if get.GetResource().Resource == "securitycontextconstraints" {
+			return true, &priv, nil
+		}
+		return false, nil, fmt.Errorf("does not exist")
+	})
+	seccli.PrependReactor("update", "*", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		get := action.(k8stesting.UpdateAction)
+		if get.GetResource().Resource == "securitycontextconstraints" {
+			return true, &priv, nil
+		}
+		return false, nil, fmt.Errorf("does not exist")
+	})
+	kc := kubeclient.NewFakeKubeclient(log, cli, seccli)
+	p := &plugin{
+		pluginConfig: template,
+		testConfig:   api.TestConfig{RunningUnderTest: true},
+		upgraderFactory: func(ctx context.Context, log *logrus.Entry, cs *api.OpenShiftManagedCluster, initializeStorageClients, disableKeepAlives bool, testConfig api.TestConfig) (cluster.Upgrader, error) {
+			return cluster.NewFakeUpgrader(ctx, log, cs, testConfig, kc, az)
+		},
+		configInterfaceFactory: config.New,
+		log:                    log,
+		now:                    time.Now,
+	}
+	err = enrich(cs)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = p.GenerateConfig(ctx, cs, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err := p.CreateOrUpdate(ctx, cs, false, getFakeDeployer(log, cs, az)); err != nil {
+		t.Errorf("plugin.CreateOrUpdate [create] error = %v", err)
+	}
+	bsc := az.StorageClient.GetBlobService()
+	updateBlobService := updateblob.NewBlobService(bsc)
+	beforeBlob, err := updateBlobService.Read()
+	// - update
+	if err := p.CreateOrUpdate(ctx, cs, true, getFakeDeployer(log, cs, az)); err != nil {
+		t.Errorf("plugin.CreateOrUpdate [update] error = %v", err)
+	}
+	// - assert that the hashes are the same
+	afterBlob, err := updateBlobService.Read()
+	if !reflect.DeepEqual(beforeBlob, afterBlob) {
+		t.Errorf("unexpected result:\n%#v\nexpected:\n%#v", spew.Sprint(beforeBlob), spew.Sprint(afterBlob))
```

Comment:
maybe use `deep.Equal` to show the differences instead? Something like...

```suggestion
		t.Errorf("unexpected blob difference: \n %#v", deep.Equal(beforeBlob, afterBlob))
```